File: merge-strings-alternately/merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string ans = "";
        int i = 0;
        for(; i < word1.size() && i < word2.size() ; i++){
            ans+=word1[i];
            ans+=word2[i];
        }
        while(i < word1.size()){
            ans+=word1[i];
            i++;
        }
        while(i < word2.size()){
            ans+=word2[i];
            i++;
        }
        
        return ans;
    }
};

File: merge-strings-alternately/merge-strings-alternately.go

func mergeAlternately(word1 string, word2 string) string {
    ans := make([]byte,len(word1) + len(word2))
    i := 0
    j := 0
    for ; i < len(word1) && i < len(word2) ; i++ {
        ans[j] = word1[i]
        ans[j+1]=word2[i]
        j += 2
    }
    
    for ; i < len(word1) ; i++ {
        ans[j] = word1[i]
        j += 1
    }
    for ; i < len(word2) ; i++ {
        ans[j] = word2[i]
        j+=1
    }
    return string(ans)
}